lib/glific/partners.ex

```diff
@@ -296,9 +298,57 @@ defmodule Glific.Partners do
     ## We need to think about a better approach to handle this one.
     Caches.remove(organization.id, ["flow_keywords_map"])
 
-    organization
-    |> Organization.changeset(attrs)
-    |> Repo.update(skip_organization_id: true)
+    with {:ok, updated_organization} <-
+           organization
+           |> Organization.changeset(attrs)
+           |> Repo.update(skip_organization_id: true) do
+      maybe_pin_newcontact_flow(
+        updated_organization.newcontact_flow_id,
+        organization.newcontact_flow_id,
+        updated_organization
+      )
+    end
+  end
+
+  @spec maybe_pin_newcontact_flow(non_neg_integer(), non_neg_integer(), Organization.t()) ::
+          {:ok, Organization.t()}
+  defp maybe_pin_newcontact_flow(nil, old_newcontact_flow_id, organization) do
+    unpin_old_newcontact_flow(old_newcontact_flow_id)
+    {:ok, organization}
+  end
+
+  defp maybe_pin_newcontact_flow(newcontact_flow_id, nil, organization) do
+    pin_new_newcontact_flow(newcontact_flow_id)
+    {:ok, organization}
+  end
+
+  defp maybe_pin_newcontact_flow(newcontact_flow_id, old_newcontact_flow_id, organization)
+       when newcontact_flow_id != old_newcontact_flow_id do
+    unpin_old_newcontact_flow(old_newcontact_flow_id)
+    pin_new_newcontact_flow(newcontact_flow_id)
+    {:ok, organization}
+  end
+
+  defp maybe_pin_newcontact_flow(_newcontact_flow_id, _old_newcontact_flow_id, organization),
+    do: {:ok, organization}
+
+  @spec unpin_old_newcontact_flow(non_neg_integer()) ::
+          {:ok, Flow.t()} | {:error, Ecto.Changeset.t()}
+  defp unpin_old_newcontact_flow(newcontact_flow_id) do
+    with false <- is_nil(newcontact_flow_id),
+         {:ok, flow} <- Flows.fetch_flow(newcontact_flow_id) do
+      Flows.update_flow(flow, %{is_pinned: false})
+    end
+  end
+
+  @spec pin_new_newcontact_flow(non_neg_integer()) ::
+          {:ok, Flow.t()} | {:error, Ecto.Changeset.t()}
+  defp pin_new_newcontact_flow(newcontact_flow_id) do
+    with {:ok, new_newcontact_flow} <- Flows.fetch_flow(newcontact_flow_id) do
+      Flows.update_flow(new_newcontact_flow, %{
+        is_pinned: true
```

Comment:
Nice work on this. 

<img width="674" alt="image" src="https://user-images.githubusercontent.com/10670690/182089904-e3c7d456-308e-42fd-8fe6-cf6cf87ff008.png">

let's fix these two test cases as well.